src/net_processing.cpp

```diff
@@ -2292,6 +2292,18 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
                 }
                 pindexWalk = pindexWalk->pprev;
             }
+            // Special case for second cmpctblock request of tip
```

Comment:
Also, maybe consider just pulling the MSG_CMPCT_BLOCK setting a bit later down up to here and just requesting the block in this if statement (possibly before the while loop above).